### PR TITLE
dot

### PR DESCRIPTION
**Pull request by CarloLucibello**
x-ref #45

### CODE REVIEW

**Comment by CarloLucibello** [comment]
julia 0.4 failing, maybe we should just drop 0.4 support

File: src/linalg/matmul.jl [hunk 1]
```diff
@@ -47,7 +47,9 @@ end
 # scale!
 # scale
 # vecdot
-# dot
+@primitive dot(x1, x2),dy,y  dy*x2  dy*x1
+gradcheckN(dot, rand(3,2), rand(3,2))
```
Comment (kirnap) [inline src/linalg/matmul.jl]: Could you please add test ``` addtestN(:dot, rand(3,2), rand(2,3))``` for your implementation

Comment (CarloLucibello) [reply src/linalg/matmul.jl]: good catch

**Comment by kirnap** [comment]
@denizyuret this PR is OK.

**Pull request merged**